### PR TITLE
publish: write Github token to Fastly service dictionary

### PR DESCRIPTION
**Pull request by stefan-guggisberg**
## Related Issues
#901

### CODE REVIEW

File: demos/full/_env [hunk 1]
```diff
@@ -8,5 +8,5 @@
 #HLX_FASTLY_AUTH=
 #HLX_FASTLY_NAMESPACE=
 
-# Github Bot
+# Github Bot & private repo support
```
Comment (tripodsan) [inline demos/full/_env]: hmmm...I'm not sure if we can use the same token for both...we need to try it out...

File: src/publish.js [hunk 2]
```diff
@@ -84,10 +84,7 @@ module.exports = function strain() {
           'Fastly Service ID is required.',
         )
         .check((args) => {
-          if (args.githubToken && args.updateBotConfig === undefined) {
-            // eslint-disable-next-line no-param-reassign
-            args.updateBotConfig = true;
-          } else if (args.updateBotConfig && !args.githubToken) {
+          if (args.updateBotConfig && !args.githubToken) {
```
Comment (tripodsan) [inline src/publish.js]: hmm...you changed the behaviour for automatically updating the bot config... why?

File: test/testPublishCli.js [hunk 3]
```diff
@@ -107,34 +107,10 @@ describe('hlx publish', () => {
...
-    sinon.assert.calledWith(mockPublish.withWskNamespace, 'hlx');
-    sinon.assert.calledWith(mockPublish.withFastlyNamespace, 'hlx'); // TODO !!
-    sinon.assert.calledWith(mockPublish.withFastlyAuth, 'secret-key');
-    sinon.assert.calledWith(mockPublish.withCustomVCLs, []);
-    sinon.assert.calledWith(mockPublish.withDispatchVersion, undefined);
-    sinon.assert.calledOnce(mockPublish.run);
+    sinon.assert.calledWith(mockPublish.withFastlyNamespace, 'hlx');
   });
 
-  it('hlx publish implicit bot config with github token', () => {
```
Comment (tripodsan) [inline test/testPublishCli.js]: why s this removed?

File: src/publish.js (hunk 2 above)
Comment (stefan-guggisberg) [reply src/publish.js]: Because `githubToken` is not used exclusively for configuring the bot but also for supporting authenticated github access. The implicit bot configuration when providing a github token was just a convenience feature and is confusing now that the github token is also used for authenticated github access.

File: test/testPublishCli.js (hunk 3 above)
Comment (stefan-guggisberg) [reply test/testPublishCli.js]: see my previous comment above.

File: demos/full/_env (hunk 1 above)
Comment (stefan-guggisberg) [reply demos/full/_env]: I agree. OTOH having to specifiy separate github tokens would be counter-intuitive and confusing.

File: src/publish.js (hunk 2 above)
Comment (tripodsan) [reply src/publish.js]: I disagree. bot config should always be updated when a github token is present. the previous behaviour was correct and should be retained.

File: test/testPublishCli.js [hunk 4]
```diff
@@ -201,17 +208,16 @@ describe('hlx publish', () => {
     sinon.assert.calledWith(mockPublish.withDispatchVersion, 'ci1');
   });
 
-  it('hlx publish handles dispatch-version with serviceid', () => {
+  it('hlx publish handles fastly-serviceid', () => {
```
Comment (tripodsan) [inline test/testPublishCli.js]: why is this changed?

File: demos/full/_env (hunk 1 above)
Comment (tripodsan) [reply demos/full/_env]: ok..let's see how this all works out
